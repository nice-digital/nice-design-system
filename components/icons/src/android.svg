<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M199.704 127.577C202.575 127.577 205.042 126.545 207.106 124.481C209.17 122.417 210.202 119.949 210.202 117.077C210.202 114.205 209.17 111.737 207.106 109.673C205.042 107.609 202.575 106.577 199.704 106.577C196.833 106.577 194.41 107.609 192.436 109.673C190.462 111.737 189.475 114.205 189.475 117.077C189.475 119.949 190.462 122.417 192.436 124.481C194.41 126.545 196.833 127.577 199.704 127.577ZM313.296 127.577C316.167 127.577 318.59 126.545 320.564 124.481C322.538 122.417 323.525 119.949 323.525 117.077C323.525 114.205 322.538 111.737 320.564 109.673C318.59 107.609 316.167 106.577 313.296 106.577C310.425 106.577 307.957 107.609 305.894 109.673C303.83 111.737 302.798 114.205 302.798 117.077C302.798 119.949 303.83 122.417 305.894 124.481C307.957 126.545 310.425 127.577 313.296 127.577ZM94.7251 177.115C102.262 177.115 108.722 179.808 114.106 185.192C119.489 190.577 122.181 197.038 122.181 204.577V320.346C122.181 328.064 119.534 334.615 114.24 340C108.947 345.385 102.442 348.077 94.7251 348.077C87.0088 348.077 80.4588 345.385 75.0753 340C69.6918 334.615 67 328.064 67 320.346V204.577C67 197.038 69.6918 190.577 75.0753 185.192C80.4588 179.808 87.0088 177.115 94.7251 177.115ZM380.052 182.231V361.538C380.052 369.795 377.181 376.795 371.438 382.538C365.696 388.282 358.787 391.154 350.712 391.154H330.523V452.269C330.523 459.987 327.832 466.538 322.448 471.923C317.065 477.308 310.515 480 302.798 480C295.082 480 288.532 477.308 283.148 471.923C277.765 466.538 275.073 459.987 275.073 452.269V391.154H237.927V452.269C237.927 459.987 235.235 466.538 229.852 471.923C224.468 477.308 217.918 480 210.202 480C202.665 480 196.205 477.308 190.821 471.923C185.437 466.538 182.746 459.987 182.746 452.269L182.477 391.154H162.558C154.303 391.154 147.304 388.282 141.562 382.538C135.819 376.795 132.948 369.795 132.948 361.538V182.231H380.052ZM317.603 73.1923C336.804 83.0641 352.147 96.8397 363.632 114.519C375.117 132.199 380.859 151.538 380.859 172.538H131.871C131.871 151.538 137.614 132.199 149.099 114.519C160.584 96.8397 176.016 83.0641 195.397 73.1923L176.286 37.9231C175.029 35.5897 175.478 33.7949 177.631 32.5385C179.964 31.4615 181.759 32 183.015 34.1538L202.396 69.6923C219.443 62.1538 237.478 58.3846 256.5 58.3846C275.522 58.3846 293.557 62.1538 310.604 69.6923L329.985 34.1538C331.241 32 333.036 31.4615 335.369 32.5385C337.522 33.7949 337.971 35.5897 336.714 37.9231L317.603 73.1923ZM446 204.577V320.346C446 328.064 443.308 334.615 437.925 340C432.541 345.385 425.991 348.077 418.275 348.077C410.738 348.077 404.278 345.385 398.894 340C393.511 334.615 390.819 328.064 390.819 320.346V204.577C390.819 196.859 393.511 190.353 398.894 185.058C404.278 179.763 410.738 177.115 418.275 177.115C425.991 177.115 432.541 179.763 437.925 185.058C443.308 190.353 446 196.859 446 204.577Z" fill="black"/>
</svg>
