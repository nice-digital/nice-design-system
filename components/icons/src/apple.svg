<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M441 358.004C434.113 380.09 423.252 402.175 408.418 424.26C385.637 458.89 362.944 476.205 340.34 476.205C331.686 476.205 319.324 473.378 303.254 467.724C288.067 462.07 274.734 459.243 263.255 459.243C252.482 459.243 239.944 462.159 225.64 467.989C211.335 473.996 199.68 477 190.673 477C163.831 477 137.253 454.12 110.94 408.359C84.9799 362.245 72 317.809 72 275.052C72 234.768 81.9777 201.728 101.933 175.933C121.889 150.49 146.966 137.769 177.164 137.769C189.879 137.769 205.508 140.419 224.05 145.72C242.416 151.02 254.602 153.671 260.606 153.671C268.553 153.671 281.179 150.667 298.486 144.66C316.499 138.653 331.775 135.649 344.313 135.649C365.328 135.649 384.136 141.391 400.736 152.876C409.919 159.236 419.102 168.07 428.285 179.378C414.334 191.216 404.268 201.64 398.087 210.651C386.608 227.259 380.869 245.546 380.869 265.511C380.869 287.419 386.961 307.12 399.146 324.611C411.332 342.103 425.283 353.234 441 358.004ZM341.399 47.131C341.399 57.9087 338.838 69.9231 333.717 83.1743C328.419 96.4255 320.207 108.617 309.082 119.748C299.546 129.288 290.009 135.649 280.473 138.829C273.939 140.773 264.756 142.275 252.924 143.335C253.454 117.009 260.341 94.3053 273.586 75.2236C286.654 56.3185 308.729 43.244 339.81 36C339.986 36.53 340.207 37.5018 340.472 38.9153C340.737 40.3287 340.958 41.3005 341.134 41.8305C341.134 42.5373 341.178 43.4207 341.267 44.4808C341.355 45.5409 341.399 46.4243 341.399 47.131Z" fill="black"/>
</svg>
