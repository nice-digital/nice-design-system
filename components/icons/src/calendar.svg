<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M81.7094 446.21H153.017V374.69H81.7094V446.21ZM168.851 446.21H248.076V374.69H168.851V446.21ZM81.7094 358.796H153.017V279.335H81.7094V358.796ZM168.851 358.796H248.076V279.335H168.851V358.796ZM81.7094 263.44H153.017V191.921H81.7094V263.44ZM263.938 446.21H343.163V374.69H263.938V446.21ZM168.851 263.44H248.076V191.921H168.851V263.44ZM359.01 446.21H430.318V374.69H359.01V446.21ZM263.938 358.796H343.163V279.335H263.938V358.796ZM176.782 144.25V72.7302C176.782 70.5793 175.996 68.7159 174.425 67.1398C172.854 65.5638 170.996 64.7758 168.851 64.7758H153.003C150.859 64.7758 149.001 65.5638 147.43 67.1398C145.858 68.7159 145.073 70.5793 145.073 72.7302V144.25C145.073 146.401 145.858 148.264 147.43 149.84C149.001 151.416 150.859 152.204 153.003 152.204H168.851C170.996 152.204 172.854 151.416 174.425 149.84C175.996 148.264 176.782 146.401 176.782 144.25ZM359.01 358.796H430.318V279.335H359.01V358.796ZM263.938 263.44H343.163V191.921H263.938V263.44ZM359.01 263.44H430.318V191.921H359.01V263.44ZM366.927 144.25V72.7302C366.927 70.5793 366.142 68.7159 364.57 67.1398C362.999 65.5638 361.141 64.7758 358.997 64.7758H343.149C341.004 64.7758 339.146 65.5638 337.575 67.1398C336.004 68.7159 335.218 70.5793 335.218 72.7302V144.25C335.218 146.401 336.004 148.264 337.575 149.84C339.146 151.416 341.004 152.204 343.149 152.204H358.997C361.141 152.204 362.999 151.416 364.57 149.84C366.142 148.264 366.927 146.401 366.927 144.25ZM462 128.355V446.21C462 454.823 458.862 462.272 452.586 468.558C446.309 474.843 438.882 477.991 430.304 478H81.6955C73.1084 478 65.6814 474.853 59.4144 468.558C53.1474 462.263 50.0092 454.814 50 446.21V128.355C50 119.743 53.1381 112.293 59.4144 106.008C65.6906 99.7222 73.1177 96.5747 81.6955 96.5655H113.391V72.7302C113.391 61.7998 117.269 52.4456 125.024 44.6673C132.779 36.8891 142.106 33 153.003 33H168.851C179.749 33 189.076 36.8891 196.831 44.6673C204.586 52.4456 208.464 61.7998 208.464 72.7302V96.5655H303.536V72.7302C303.536 61.7998 307.414 52.4456 315.169 44.6673C322.924 36.8891 332.251 33 343.149 33H358.997C369.894 33 379.221 36.8891 386.976 44.6673C394.731 52.4456 398.609 61.7998 398.609 72.7302V96.5655H430.304C438.892 96.5655 446.319 99.7129 452.586 106.008C458.853 112.303 461.991 119.752 462 128.355Z" fill="black"/>
</svg>
