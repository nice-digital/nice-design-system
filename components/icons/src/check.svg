<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M184.5 407.794L38.5752 270.073C29.8083 261.799 29.8083 248.384 38.5752 240.109L70.3235 210.145C79.0905 201.87 93.3059 201.87 102.073 210.145L200.375 302.919L410.927 104.206C419.694 95.9315 433.91 95.9315 442.676 104.206L474.425 134.17C483.192 142.444 483.192 155.859 474.425 164.134L216.25 407.795C207.482 416.069 193.267 416.069 184.5 407.794V407.794Z" fill="black"/>
</svg>
