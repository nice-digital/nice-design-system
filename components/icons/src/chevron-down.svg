<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M478.152 174C478.152 184.091 474.627 192.661 467.576 199.712L281.576 385.712C274.333 392.955 265.667 396.576 255.576 396.576C245.293 396.576 236.723 392.955 229.864 385.712L43.864 199.712C36.6213 192.853 33 184.283 33 174C33 163.909 36.6213 155.243 43.864 148L65 126.576C72.424 119.525 81.0907 116 91 116C101.091 116 109.661 119.525 116.712 126.576L255.576 265.44L394.44 126.576C401.491 119.525 410.061 116 420.152 116C430.061 116 438.728 119.525 446.152 126.576L467.576 148C474.627 155.424 478.152 164.091 478.152 174Z" fill="black"/>
</svg>
