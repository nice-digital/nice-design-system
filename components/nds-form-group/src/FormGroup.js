import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "./../scss/form-group.scss";

export const FormGroup = props => {
	const { groupError, inline, legend, children, hint, name, ...rest } = props;

	const clonedChildren = React.Children.map(children, child => {
		return React.cloneElement(child, {
			name,
			inline,
			...rest
		});
	});

	const classes = classnames({
		"form-group": true,
		"form-group--no-legend": legend ? false : true
	});

	return (
		<fieldset className={classes}>
			{legend && <legend className="form-group__legend">{legend}</legend>}
			{groupError && <p className="form-group__error-message">{groupError}</p>}
			{hint && <p className="form-group__hint">{hint}</p>}
			{clonedChildren}
		</fieldset>
	);
};

FormGroup.propTypes = {
	children: PropTypes.node.isRequired,
	legend: PropTypes.node,
	name: PropTypes.string,
	hint: PropTypes.node,
	inline: PropTypes.bool,
	groupError: PropTypes.string
};
