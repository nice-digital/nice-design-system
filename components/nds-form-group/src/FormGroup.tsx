import React from "react";
import classnames from "classnames";
import "./../scss/form-group.scss";

export interface FormGroupProps {
	[prop: string]: unknown;
	inline?: boolean;
	name?: string;
	hint?: React.ReactNode;
	legend?: string;
	children: React.ReactNode;
	groupError?: string | boolean;
}

export const FormGroup: React.FC<FormGroupProps> = (props: FormGroupProps) => {
	const { groupError, inline, legend, children, hint, name, ...rest } = props;

	const clonedChildren = React.Children.map(children, (child) => {
		return React.cloneElement(child as React.ReactElement, {
			name,
			inline,
			...rest
		});
	});

	const classes = classnames({
		"form-group": true,
		"form-group--no-legend": legend ? false : true
	});

	return (
		<fieldset
			className={classes}
			data-component={`form-group${inline ? "--inline" : ""}`}
		>
			{legend && <legend className="form-group__legend">{legend}</legend>}
			{groupError && <p className="form-group__error-message">{groupError}</p>}
			{hint && <p className="form-group__hint">{hint}</p>}
			{clonedChildren}
		</fieldset>
	);
};
