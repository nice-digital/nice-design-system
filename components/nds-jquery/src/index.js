import pluginAutoLoader from "./plugin-autoloader";
import Tabs from "./tabs.js";
import InPageNav from "./in-page-nav.js";
import Tracker from "./tracker";
import pluginizr from "./pluginizr";
import eventr from "./eventr";
import breakpoints from "../../nds-core/src/breakpoints";
import utils from "../../nds-core/src/utils";

let nice = {
	init: (el: $) => {

		// Load all component modules from this directory automatically, see http://stackoverflow.com/a/31770875/486434
		// But exclude test files http://stackoverflow.com/a/30372240
		pluginAutoLoader.load(require.context("./../components/", true, /^((?!test\.).)*\.js$/im));
		pluginAutoLoader.load(require.context("./", true, /^((?!test\.).)*\.js$/im));

		pluginAutoLoader.findPlugins(el);
	},
	// Components
	InPageNav: InPageNav,
	Tabs: Tabs,
	Tracker: Tracker,
	// Utils
	pluginAutoLoader: pluginAutoLoader,
	pluginizr: pluginizr,
	eventr: eventr,
	breakpoints,
	utils
};

$.fn.nice = function() {
	nice.init(this);
};

// Export to global namespace for precompiled usage
window.NICE = window.NICE || {};
window.NICE.nice = nice;

export {
	nice as default,
	// Components
	InPageNav,
	Tabs,
	Tracker,
	// Utils
	pluginAutoLoader,
	pluginizr,
	eventr,
	breakpoints,
	utils
};
