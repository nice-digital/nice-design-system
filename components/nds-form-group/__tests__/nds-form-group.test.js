"use strict";

import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { FormGroup } from "./../src/FormGroup";

const props = {
	legend: "This is the legend text",
	hint: "This is the hint text",
	groupError: "This is the error text",
	inline: false,
	name: "group-name"
};

describe("FormGroup", () => {
	it("should render without crashing", () => {
		const wrapper = shallow(
			<FormGroup {...props}>
				<span>Render</span>
			</FormGroup>
		);
		expect(wrapper).toHaveLength(1);
	});

	it("should match the snapshot when no props are passed", () => {
		const wrapper = shallow(
			<FormGroup>
				<span>Render</span>
			</FormGroup>
		);
		expect(toJson(wrapper, { noKey: true })).toMatchSnapshot();
	});

	it("should match the snapshot when all possible props are passed", () => {
		const wrapper = shallow(
			<FormGroup {...props}>
				<span>Render</span>
			</FormGroup>
		);
		expect(toJson(wrapper, { noKey: true })).toMatchSnapshot();
	});

	it("should pass any additional props down to the cloned child components", () => {
		const localProps = Object.assign({}, props);
		localProps.inline = true;
		const wrapper = shallow(
			<FormGroup {...localProps}>
				<span>Render</span>
			</FormGroup>
		);
		const child = wrapper.find("span");
		expect(child.props()["name"]).toEqual("group-name");
		expect(child.props()["inline"]).toEqual(true);
	});
});
