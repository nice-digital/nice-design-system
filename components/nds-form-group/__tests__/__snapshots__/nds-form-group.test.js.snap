// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should match the snapshot when all possible props are passed 1`] = `
<fieldset
  className="form-group"
>
  <legend
    className="form-group__legend"
  >
    This is the legend text
  </legend>
  <p
    className="form-group__error-message"
  >
    This is the error text
  </p>
  <p
    className="form-group__hint"
  >
    This is the hint text
  </p>
  <span
    inline={false}
    name="group-name"
  >
    Render
  </span>
</fieldset>
`;

exports[`FormGroup should match the snapshot when no props are passed 1`] = `
<fieldset
  className="form-group form-group--no-legend"
>
  <span>
    Render
  </span>
</fieldset>
`;
